dgraph/cmd/zero/http.go

```diff
@@ -231,9 +234,44 @@ func (st *state) getState(w http.ResponseWriter, r *http.Request) {
 	}
 }
 
+func (s *Server) zeroHealth(ctx context.Context) (*api.Response, error) {
+	if ctx.Err() != nil {
+		return nil, ctx.Err()
+	}
+
+	healthAll := []pb.HealthInfo{pb.HealthInfo{
+		Instance:    "zero",
+		Address:     x.WorkerConfig.MyAddr,
+		Status:      "healthy",
+		Version:     x.Version(),
+		Uptime:      int64(time.Since(x.WorkerConfig.StartTime) / time.Second),
+		LastEcho:    time.Now().Unix(),
+	}}
+
+	jsonOut, err := json.Marshal(healthAll)
+	if err != nil {
+		return nil, errors.Errorf("Unable to Marshal. Err %v", err)
+	}
+	return &api.Response{Json: jsonOut}, nil
+}
+
 func (st *state) pingResponse(w http.ResponseWriter, r *http.Request) {
+	var err error
 	x.AddCorsHeaders(w)
 
-	w.WriteHeader(http.StatusOK)
-	_, _ = w.Write([]byte("OK"))
+	switch r.Header.Get("Accept") {
+		case "application/json":
+			var resp *api.Response
```

Comment:
same, no need for declaration

Reply:
Actually, resp is used at line #271, if I restrict its declaration to 'if' block, the compiler would throw error, and since zeroHealth()'s return is multi-valued, I think, we need to have both, resp and err as they are. But, this is what I can do- move statement (var err error) at line# 259 to line# 264 just to reduce it's scope.

Reply:
then you can just move the statement outside the if block